stdlib/private/OSLog/CMakeLists.txt

```diff
@@ -6,9 +6,14 @@ add_swift_target_library(swiftOSLogPrototype
   OSLog.swift
   OSLogMessage.swift
 
-  SWIFT_MODULE_DEPENDS_IOS Darwin os
-  SWIFT_MODULE_DEPENDS_OSX Darwin os
-  SWIFT_MODULE_DEPENDS_TVOS Darwin os
-  SWIFT_MODULE_DEPENDS_WATCHOS Darwin os
+  # This module has code that must be constant evaluated at compile time in the
+  # context of the client. Optimizing this module interferes with the constant
+  # evaluator by inlining/eliminating functions with @_semantics attributes.
+  SWIFT_COMPILE_FLAGS "-Onone"
```

Comment:
Out of curiosity, would `@inline(__never)` applied to all these functions do the trick?

Reply:
The actual problem is inlining function calls in the os_log implementation and not inlining the os_log functions into their callers. The only real problem so far is inlining the calls to `String.+=` used in the construction of the format string. To address this, we can either annotate: [String.+=](https://github.com/apple/swift/blob/d044696b30ca70a0b894812b468b659096af107f/stdlib/public/core/String.swift#L573) function as `@inline(never)`. But that will have a more sweeping effect.  Another alternative I just realized is to forward the same semantic attribute to the function [String.append](https://github.com/apple/swift/blob/d044696b30ca70a0b894812b468b659096af107f/stdlib/public/core/StringRangeReplaceableCollection.swift#L128) called by `+=`, which is not marked @inlinable.  Let me see if I could change to that and remove the -Onone requirement here.

Reply:
Actually, I found that, while the string.append problem I mentioned above can be fixed,  enabling optimization on this module makes string internal structures like `_StringGuts` visible. It seems to be reading and loading data from them. The evaluator is not aware of those internal details (which are even outside the constant evaluable fragment) and is not able to fold the format string.